### PR TITLE
Add BigQuery BIReservation Resource

### PR DESCRIPTION
**Pull request by obada-ab**
Add BigQuery BIReservation Resource.

Documentation: https://cloud.google.com/bigquery/docs/reference/reservations/rpc/google.cloud.bigquery.reservation.v1#bireservation

This resource is a singleton and only supports update/get operations, so some custom code was added:
* `pre_create` for adding `updateMask` during creation
* `custom_delete` where the  fields are cleared
* `test_check_destroy` for checking whether the fields have been cleared, because the resource cannot be deleted

Fixes https://github.com/hashicorp/terraform-provider-google/issues/15446

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_bigquery_bi_reservation
```

### CODE REVIEW

**Comment by shuyama1** [comment]
Sorry, I don't think I'll be able to review this PR today as it's a busy week for oncall. I'll be out for the next couple of days, so finding a new reviewer

File: mmv1/products/bigqueryreservation/BiReservation.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    description: |
+      The last update timestamp of a reservation.
+
+      A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+  - !ruby/object:Api::Type::Integer
+    name: 'size'
+    description: |
+      Size of a reservation, in bytes.
+  - !ruby/object:Api::Type::Array
+    name: 'preferredTables'
```
Comment (hao-nan-li) [inline mmv1/products/bigqueryreservation/BiReservation.yaml:74]: Mark this field with required?

**Review by hao-nan-li — COMMENTED** [review]
Could you add an update test?

File: mmv1/products/bigqueryreservation/BiReservation.yaml (hunk 1 above)
Comment (obada-ab) [reply mmv1/products/bigqueryreservation/BiReservation.yaml:73]: This field should be optional, changed subfields as well

**Comment by obada-ab** [comment]
@hao-nan-li added an update test

**Review by hao-nan-li — APPROVED** [review]

**Pull request merged**